### PR TITLE
samples: drivers: led_strip: add nucleo_h563zi support

### PR DESCRIPTION
**Pull request by jeppenodgaard**
Add config and overlay files for nucleo_h563zi evaluation kit in apa102 and ws2812 samples.

### CODE REVIEW

File: samples/drivers/led_ws2812/boards/nucleo_h563zi.overlay [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	led_strip: ws2812@0 {
+		compatible = "worldsemi,ws2812-spi";
+
+		/* SPI */
+		reg = <0>; /* ignored, but necessary for SPI bindings */
+		spi-max-frequency = <DT_FREQ_M(4)>;
+		frame-format = <32768>; /* SPI_FRAME_FORMAT_TI */
+
+		/* WS2812 */
+		chain-length = <720>; /* arbitrary; change at will */
```
Comment (thedjnK) [inline samples/drivers/led_ws2812/boards/nucleo_h563zi.overlay:20]: 720 is a bit much, 16 is used by other boards

Comment (simonguinot) [reply samples/drivers/led_ws2812/boards/nucleo_h563zi.overlay:20]: Excellent ! Does it work ? :)

Comment (jeppenodgaard) [reply samples/drivers/led_ws2812/boards/nucleo_h563zi.overlay]: Turns out the LEDs are in culsters of 15 LEDs, so it is 720 LEDs / 15 = 48.
I've updated the code.

**Review by simonguinot — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]
Ok but why? These boards don't seem to have such a device on board, is this for a shield or a custom setup or something? Then it should not be part of the sample.

**Comment by erwango** [comment]
> Ok but why? These boards don't seem to have such a device on board, is this for a shield or a custom setup or something? Then it should not be part of the sample.

Agreed. We're not describing a specific board component here and potentially we can add a infinite number of similar custom configurations. I don't think we should go in that direction. Having few samples is required, but there are few already and this one doesn't add any specific value.

**Comment by jeppenodgaard** [comment]
> > Ok but why? These boards don't seem to have such a device on board, is this for a shield or a custom setup or something? Then it should not be part of the sample.
> 
> Agreed. We're not describing a specific board component here and potentially we can add a infinite number of similar custom configurations. I don't think we should go in that direction. Having few samples is required, but there are few already and this one doesn't add any specific value.

Makes sense.

**Pull request closed without merging**